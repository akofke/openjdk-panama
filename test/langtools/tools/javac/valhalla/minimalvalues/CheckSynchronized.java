/*
 * @test /nodynamiccopyright/
 * @summary Check behavior of synzhronized key word on value instances and methods.
 * @modules jdk.incubator.mvt
 * @compile/fail/ref=CheckSynchronized.out --should-stop:ifError=PARSE -XDrawDiagnostics -Werror -Xlint:values  CheckSynchronized.java
 */
@jdk.incubator.mvt.ValueCapableClass
final class CheckSynchronized {
    synchronized void foo() { // <<-- ERROR, no monitor associated with `this'
    }
    void goo() {
        synchronized(this) {} // <<-- ERROR, no monitor associated with `this'
    }
    synchronized static void zoo(CheckSynchronized cs) { // OK, static method.
        synchronized(cs) {    // <<-- ERROR, no monitor associated with value instance.
        }
    }
}
