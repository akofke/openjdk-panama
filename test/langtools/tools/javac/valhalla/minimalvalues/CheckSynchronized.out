- compiler.warn.incubating.modules: jdk.incubator.mvt
- compiler.err.warnings.and.werror
CheckSynchronized.java:9:23: compiler.warn.mod.not.allowed.here: synchronized
CheckSynchronized.java:12:9: compiler.warn.type.found.req: CheckSynchronized, (compiler.misc.type.req.ref)
CheckSynchronized.java:15:9: compiler.warn.type.found.req: CheckSynchronized, (compiler.misc.type.req.ref)
1 error
4 warnings
